### PR TITLE
privilege: fix privilege check of `CREATE ROLE` and `DROP ROLE`

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
In MySQL, user with `CREATE USER` privilege can execute `CREATE ROLE xxx` or `DROP ROLE xxx`.  But TiDB has bug on this check.

```
mysql> show grants;
+------------------------------------------+
| Grants for User                          |
+------------------------------------------+
| GRANT Create User ON *.* TO 'tester'@'%' |
+------------------------------------------+
1 row in set (0.00 sec)

mysql> create role r4;
ERROR 1227 (42000): Access denied; you need (at least one of) the CREATE ROLE privilege(s) for this operation
mysql> drop role r1;
ERROR 1227 (42000): Access denied; you need (at least one of) the DROP ROLE privilege(s) for this operation
mysql>
```

### What is changed and how it works?
Add check for `CREATE USER` privilege. If users have `CREATE USER` privilege, they can `CREATE ROLE` and `DROP ROLE` 

### Check List 

Tests 

 - Unit test

Code changes

 - None

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

Release note

 - fix bug of privilege check on `CREATE ROLE`

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: executor/simple.go [hunk 1]
```diff
@@ -829,8 +835,11 @@ func (e *SimpleExec) executeDropUser(s *ast.DropUserStmt) error {
 		}
 		activeRoles := e.ctx.GetSessionVars().ActiveRoles
 		if !checker.RequestVerification(activeRoles, mysql.SystemDB, mysql.UserTable, "", mysql.DeletePriv) {
-			if s.IsDropRole && !checker.RequestVerification(activeRoles, "", "", "", mysql.DropRolePriv) {
-				return core.ErrSpecificAccessDenied.GenWithStackByArgs("DROP ROLE")
+			if s.IsDropRole {
+				if !checker.RequestVerification(activeRoles, "", "", "", mysql.DropRolePriv) &&
```
Comment (tiancaiamao) [inline executor/simple.go]: Here is `&&`, that means if a user has the `CreateUser` privilege **or** `DropRole` privilege, the can execute the drop role operation.

Comment (imtbkcat) [reply executor/simple.go]: right, if a user has the CreateUser privilege or DropRole privilege, the can execute the drop role operation.

**Pull request merged**